src/smartstream/derive/src/generator/filter.rs

```diff
@@ -21,15 +21,13 @@ pub fn generate_filter_smartstream(func: &SmartStreamFn) -> TokenStream {
                 let input_data = Vec::from_raw_parts(ptr, len, len);
                 let mut smartstream_input = fluvio_smartstream::dataplane::smartstream::SmartStreamInput::default();
                 if let Err(_err) = fluvio_smartstream::dataplane::core::Decoder::decode(&mut smartstream_input, &mut std::io::Cursor::new(input_data), 0) {
-                    // return fluvio_smartstream::ENCODING_ERROR;
-                    return -11;
+                    return fluvio_smartstream::dataplane::smartstream::SmartStreamInternalError::DecodingBaseInput as i32;
```

Comment:
this is verbose. can you import `SmartStreamInternalError` at top?

Reply:
It is bad practice to put imports in derive macros, because then those imports will pollute the user's code when they get imported. If you look at the macro expansions for a lot of popular crates they all use full crate paths like this.

Reply:
Not necessary.  You can put that in the beginning of the fn or block.

Reply:
Ah, looking at this closer, we would not be mixing generated code into the user code anyway so it should be fine 👍